docs/cookbook/create-advanced-grid.md

```diff
@@ -0,0 +1,463 @@
+```php
+// src/Shopsys/ShopBundle/Grid/Salesman/SalesmanGridInlineEdit.php
+
+namespace Shopsys\ShopBundle\Grid\Salesman;
+
+use Shopsys\FrameworkBundle\Component\Grid\InlineEdit\AbstractGridInlineEdit;
+use Shopsys\ShopBundle\Form\Admin\SalesmanFormType;
+use Shopsys\ShopBundle\Model\Salesman\SalesmanDataFactory;
+use Shopsys\ShopBundle\Model\Salesman\SalesmanFacade;
+use Symfony\Component\Form\FormFactoryInterface;
+
+class SalesmanGridInlineEdit extends AbstractGridInlineEdit
+{
+    /**
+     * @var \Shopsys\ShopBundle\Grid\Salesman\SalesmanGridFactory
+     */
+    private $salesmanGridFactory;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\Salesman\SalesmanFacade
+     */
+    private $salesmanFacade;
+
+    /**
+     * @var \Symfony\Component\Form\FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\Salesman\SalesmanDataFactory
+     */
+    private $salesmanDataFactory;
+
+    public function __construct(
+        SalesmanGridFactory $salesmanGridFactory,
+        SalesmanFacade $salesmanFacade,
+        FormFactoryInterface $formFactory,
+        SalesmanDataFactory $salesmanDataFactory
+    ) {
+        parent::__construct($salesmanGridFactory);
+        $this->salesmanGridFactory = $salesmanGridFactory;
+        $this->salesmanFacade = $salesmanFacade;
+        $this->formFactory = $formFactory;
+        $this->salesmanDataFactory = $salesmanDataFactory;
+    }
+
+    /**
+     * @param int|null $salesmanId
+     * @return \Symfony\Component\Form\FormInterface
+     */
+    public function getForm($salesmanId)
+    {
+        if ($salesmanId === null) {
+            $salesmanData = $this->salesmanDataFactory->create();
+        } else {
+            $salesman = $this->salesmanFacade->getById($salesmanId);
+            $salesmanData = $this->salesmanDataFactory->createFromSalesman($salesman);
+        }
+
+        return $this->formFactory->create(SalesmanFormType::class, $salesmanData);
+    }
+
+    /**
+     * @param int $salesmanId
+     * @param \Shopsys\ShopBundle\Model\Salesman\SalesmanData $salesmanData
+     */
+    protected function editEntity($salesmanId, $salesmanData)
+    {
+        $this->salesmanFacade->edit($salesmanId, $salesmanData);
+    }
+
+    /**
+     * @param \Shopsys\ShopBundle\Model\Salesman\SalesmanData $salesmanData
+     * @return int
+     */
+    protected function createEntityAndGetId($salesmanData)
+    {
+        $salesman = $this->salesmanFacade->create($salesmanData);
+
+        return $salesman->getId();
+    }
+}
+```
+The new class must be registered in `src/Shopsys/ShopBundle/Resources/config/services.yml`:
+```yaml
+Shopsys\ShopBundle\Grid\Salesman\SalesmanGridInlineEdit: ~
+```
+
+### 1.7 Use `SalesmanGridInlineEdit` in `SalesmanController`
+To make the salesman grid inline editable now, we need to use the `SalesmanGridInlineEdit::getGrid` method to get the grid
+instead of calling `SalesmanGridFactory::create` method directly:
```

Comment:
some random unwanted line break

Reply:
but also the sentence is kind of strange

Reply:
hm, I think it is strange because the actual fact is kind of strange :smile: